### ISSUE TITLE
can I use this with page_cache?

### ISSUE THREAD

**Issue by patroqueeet** [litchfield/django-fastsitemaps#8]
if so, how?

**Comment by patroqueeet** [comment]
no update?

**Comment by litchfield** [comment]
Not sure it would make sense to do that? If it's cached, it'll be coming straight from the cache.

**Comment by tingletech** [comment]
I think this question might be related to the issue I'm struggling with at the moment.

I have [django per-site cache](https://docs.djangoproject.com/en/1.9/topics/cache/#the-per-site-cache) turned on.  The middleware throws a pickle error unless I use the `never_cache` decorator.

```
Exception Type: PicklingError at /sitemap-static.xml
Exception Value: Can't pickle <type 'generator'>: attribute lookup __builtin__.generator failed
```

http://dpaste.com/0B8BFNM

I don't think I want "never_cache" b/c then cloud front won't cache the pages.

I guess I have to switch to per-view cache

**Comment by litchfield** [comment]
Yes unfortunately streaming doesn't play well with caching- it doesn't really make sense anyway, except for the benefit of less memory usage?

If you're using a CDN, do you really need this?

**Comment by patroqueeet** [comment]
at least for me it is desirable as rendering the pages on request took too much time and my nginx timed out. regenerate once a day and then cache for the many GoogleBot accesses seemed to be reasonable to me. for now I switched back to regular django sitemaps with caching and my SEO rank responded positively to that ...

**Comment by tingletech** [comment]
Maybe I don't need this (or maybe I don't understand what this does), but in my case I've got over 500,000 pages on my site which are backed by a sorl index.  Somewhere else in my codebase, I have a generator that will page the solr results with `cursorMark`.  Looking at `django.contrib.sitemaps` I can't see how that would be able to work for this -- and I don't want to have a special process to generate sitemaps if I can avoid it.
